source/common/conn_pool/conn_pool_base.cc

```diff
@@ -77,7 +77,7 @@ bool ConnPoolImplBase::shouldCreateNewConnection(float global_preconnect_ratio)
   // upstream selection logic may result in bypassing this upstream entirely.
   // If an Envoy user wants preconnecting for degraded upstreams this could be
   // added later via extending the preconnect config.
-  if (host_->health() != Upstream::Host::Health::Healthy) {
+  if (host_->coarseHealth() != Upstream::Host::Health::Healthy) {
```

Comment:
Not in this PR, but in a followup, should we adjust as many of these `coarseHealth` to `healthStatus` as we can? I'm hoping that we can generally get rid of most of the old-style, since it's a bit confusing to read code and wonder why we need to treat the two cases differently. The alternative is to move everything over at once, not sure what would break though.

Reply:
If we just want to remove this `coarseHealth` method from `Host`, it's easy for us to do this. We can use a simple util function + `healthStatus` to replace this method in the future.

But it's a little hard to complete remove this simple status abstraction from Envoy.

Reply:
Sure. I think that would be a good place to get to, where we just do the abstraction where we need it.

Reply:
I will add a todo to my task list. This wouldn't be too hard.